cmd/dsse/dsse.go

```diff
@@ -0,0 +1,319 @@
+		if err != nil {
+			return nil, fmt.Errorf("unable to get absolute path of '%s': %w", keyfilePath, err)
+		}
+
+		_, err = os.Stat(keyfilePath)
+		if err != nil {
+			return nil, fmt.Errorf("unable to verify keyfile': %w", err)
+		}
+	}
+
+	return &OptionsAndArgs{
+		keyfilePath: keyfilePath,
+		dssePath:    dssePath,
+		command:     command,
+	}, nil
+}
+
+func load(DSSEPath string) (*gitrelease.Envelope, error) {
+	data, err := os.ReadFile(DSSEPath)
+	if err != nil {
+		return nil, err
+	}
+
+	envelope := gitrelease.Envelope{}
+	err = json.Unmarshal(data, &envelope)
+	if err != nil {
+		return nil, err
+	}
+
+	return &envelope, nil
+}
+
+func getPAE(envelope *gitrelease.Envelope) ([]byte, error) {
+	payload, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	preAuthenticationEncoding, err := gitrelease.PreAuthenticationEncoding(payload)
+	if err != nil {
+		return nil, err
+	}
+
+	return preAuthenticationEncoding, nil
+}
+
+func signAndWrite(DSSEPath string, envelope *gitrelease.Envelope, keyfilePath string) error {
+	existing, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return err
+	}
+
+	err = printPayload(envelope)
+	if err != nil {
+		return err
+	}
+
+	signature, err := sign(existing, keyfilePath)
+	if err != nil {
+		return err
+	}
+
+	envelope.Signatures = append(envelope.Signatures, gitrelease.Signature{
+		Sig: signature,
+	})
+
+	output, err := json.Marshal(envelope)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(DSSEPath, output, os.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func printPayload(envelope *gitrelease.Envelope) error {
+	payload, err := base64.StdEncoding.DecodeString(envelope.Payload)
+	if err != nil {
+		return err
+	}
+	var t any
+	err = json.Unmarshal(payload, &t)
+	if err != nil {
+		return err
+	}
+
+	formatted, err := json.MarshalIndent(t, "  ", "  ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(formatted))
+
+	return nil
+}
+
+func sign(payload []byte, keyfilePath string) (string, error) {
+	preAuthenticationEncoding, err := gitrelease.PreAuthenticationEncoding(payload)
+	if err != nil {
+		return "", fmt.Errorf("unable to create signature payload: %w", err)
+	}
+
+	tmpFile, err := os.CreateTemp("", "payload")
+	if err != nil {
+		return "", fmt.Errorf("unable to create temporary file: %w", err)
+	}
+
+	_, err = tmpFile.Write(preAuthenticationEncoding)
+	if err != nil {
+		return "", fmt.Errorf("unable to write to temporary file: %w", err)
+	}
+	err = tmpFile.Sync()
+	if err != nil {
+		return "", fmt.Errorf("unable to sync temporary file: %w", err)
+	}
+
+	command := []string{"ssh-keygen", "-Y", "sign", "-n", sshNamespace, "-f", keyfilePath, tmpFile.Name()}
+
+	fmt.Printf("Executing command: %s\n", command)
+	cmd := exec.Command(command[0], command[1:]...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	err = cmd.Run()
+	if err != nil {
+		return "", err
+	}
+
```

Comment:
Use `defer` for cleanup.
```suggestion
	err = cmd.Run()
	if err != nil {
		return "", err
	}
        
        defer os.Remove(tmpFile.Name() + ".sig")
```